examples/type-conversion/type_conversion.bal

```diff
@@ -1,71 +1,106 @@
+// The conversion would return an error if non-`anydata` or an incompatible value is found.
+function convertAnyMapToPerson(map<any> m) {
+    // Attempt creating a new value of type `Person` from the `map<any>` typed `m` value, without changing `m`'s
+    // inherent type.
+    Person|error res = Person.convert(m);
+    if (res is Person) {
+        // If the conversion is successful, print the `name` field.
+        io:println("map<any> to Person, name: ", res["name"]);
+    } else {
+        io:println("error occurred on conversion");
+    }
+}
 
-    // The `convert()` can be used to explicity convert simple values as below.
+// Function to convert simple basic types using the `.convert()` built-in method.
+function convertSimpleBasicTypes() {
     string s1 = "45";
     string s2 = "abc";
     string s3 = "true";
     float f = 10.2;
     any a = 3.14;
-    
-    int|error intVal1 = int.convert(s1);
-
-    int|error intVal2 = int.convert(s2);
-
-    int intVal3 = int.convert(f);
 
-    boolean|error b = boolean.convert(s3);
-
-    float|error af = float.convert(a);
-
-    if (intVal1 is int) {
-        io:println("Int value 1 : " + intVal1);
+    // `string` to `int` conversion is unsafe, since the `string` may not be convertible to `int`.
+    int|error res1 = int.convert(s1);
+    if (res1 is int) {
+        io:println("int value: ", res1);
     } else {
-        io:println("Error: " + intVal1.reason());
+        io:println("error: ", res1.detail().message);
     }
 
-    if (intVal2 is int) {
-        io:println("Int value 2 : " + intVal2);
+    res1 = int.convert(s2);
+    if (res1 is int) {
+        io:println("int value: ", res1);
     } else {
-        io:println("Error: " + intVal2.reason());
+        io:println("error: ", res1.detail().message);
     }
 
-    io:println("Int value 3 : " + intVal3);
+    // A `float` to `int` conversion can result in some of the information getting lost.
+    // However, this conversion is always safe since there is a corresponding `int` representation
+    // for all `float` values except for `NaN` or `infinite` float values, in which case the
+    // the conversion attempt will result in a panic.
+    int intVal = int.convert(f);
+    io:println("int value: ", intVal);
 
-    if (b is boolean) {
-        io:println("Boolean value : " + b);
-    } else {
-        io:println("Error: " + b.reason());
-    }
+    // A `string` to `boolean` conversion is always safe. The `string` value `true` (ignoring case)
+    // evaluates to the `boolean` value `true`, while any other `string` is converted to the
+    // `boolean` value `false`.
+    boolean b = boolean.convert(s3);
+    io:println("boolean value: ", b);
 
-    if (af is float) {
-        io:println("Float value : " + af);
+    // A simple basic typed value held in an `any` typed variable can also be converted to its inherent
+    // type using the `.convert()` method. This attempt is unsafe, since the value may not be compatible
```

Comment:
method or function?

Reply:
We've used method for built-in methods (same reason for https://github.com/ballerina-platform/ballerina-lang/pull/12814/files#r242126073).

Ref: https://docs.google.com/presentation/d/1ZnykKSai8WFJzT6GMIuj1s-6bMwGYaawm_Q1E4_PBa4/edit#slide=id.g424ccecdb3_0_55